nucypher/blockchain/eth/agents.py

```diff
@@ -372,11 +371,10 @@ def get_all_stakes(self, staker_address: str):
             yield self.get_substake_info(staker_address=staker_address, stake_index=stake_index)
 
     @validate_checksum_address
-    def deposit_tokens(self, amount: int, lock_periods: int, sender_address: str):
-        """Send tokens to the escrow from the staker's address"""
-        contract_function = self.contract.functions.deposit(amount, lock_periods)
-        receipt = self.blockchain.send_transaction(contract_function=contract_function,
-                                                   sender_address=sender_address)
+    def deposit_tokens(self, amount: int, lock_periods: int, staker_address: str, sender_address: str) -> dict:
```

Comment:
Maybe add default behaviour for case when `staker_address` is `None`? (staker_address=sender_address)

Reply:
:heavy_check_mark: - Although did it the other way around to remain consistent with the rest of the module (`staker_address` is required, `sender_address` is optional).